bodo/pandas/_executor.h

```diff
@@ -23,19 +23,25 @@ class Executor {
         PhysicalPlanBuilder builder;
         builder.Visit(*plan);
         pipelines = std::move(builder.finished_pipelines);
-
-        if (builder.active_pipeline != nullptr) {
-            pipelines.push_back(builder.active_pipeline->BuildEnd(out_schema));
-        }
+        assert(builder.active_pipelines.size() == 1);
+        pipelines.push_back(
+            builder.active_pipelines.top().first->BuildEnd(out_schema));
     }
 
     /**
-     * @brief Execute the plan and return the result (placeholder for now).
+     * @brief Execute the plan and return the result.
      */
     std::shared_ptr<table_info> ExecutePipelines() {
-        // TODO: support multiple pipelines
-        pipelines[0]->Execute();
-
-        return pipelines[0]->GetResult();
+        // We must execute all pipelines once.
+        // Though there are dependencies between them
+        // the method of construction should guarantee that
+        // earlier items in the vector are dependencies to
+        // later items so we can still process them in order.
+        for (size_t i = 0; i < pipelines.size(); ++i) {
```

Comment:
Why do we need to keep track of dependencies in pipelines? Does DuckDB do it? Pipelines just need to run one at a time in order in regular SQL I believe.

Reply:
Here is the way I see it.  Some pipelines do depend on the results of other pipelines (if they didn't then they'd be separate queries).  So, those pipelines need a way to get the results from their dependent pipelines.  There's probably multiple ways of arranging to do that but we already had the ability to query a pipeline to get its results (i.e., GetResults) so if you just have a reference to the pipelines you are dependent on then you can get their results as needed.